File: api-integration/capture-api/basic-batch.js
// Load environment variables from .env file
require('dotenv').config();

const headers = {
    "Content-Type": "application/json",
};

const body = {
    "api_key": process.env.POSTHOG_API_KEY,
    "batch": [
        {
            "event": "batched_event",
            "properties" : {
                "distinct_id": "test-user-javascript",
                "number_in_batch": 1
            }
        },
        {
            "event": "batched_event",
            "properties" : {
                "distinct_id": "test-user-javascript",
                "number_in_batch": 2
            }
        }
    ]
};

const url = `${process.env.POSTHOG_API_HOST}/batch/`;

fetch(url, {
    method: 'POST',
    headers: headers,
    body: JSON.stringify(body)
})
.then(response => response.json())
.then(data => console.log(data))
.catch(error => console.error('Error:', error));

File: api-integration/capture-api/basic-batch.py
import requests
import os
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

url = f"{os.getenv('POSTHOG_API_HOST')}/batch/"

headers = {
    "Content-Type": "application/json",
}

body = {
    "api_key": os.getenv('POSTHOG_API_KEY'),
    "batch": [
        {
            "event": "batched_event",
            "properties" : {
                "distinct_id": "test-user-python",
                "number_in_batch": 1
            }
        },
        {
            "event": "batched_event",
            "properties" : {
                "distinct_id": "test-user-python",
                "number_in_batch": 2
            }
        }
    ]
}

response = requests.post(url, headers=headers, json=body)

print(response.json())
